schema_salad/ref_resolver.py

```diff
@@ -522,36 +531,52 @@ def check_file(self, fn):  # type: (Union[str, unicode]) -> bool
         else:
             return False
 
-    def validate_link(self, field, link):
+    def validate_link(self, field, link, docid):
         # type: (str, Union[str, unicode, List[str], Dict[str, Any]]) -> bool
         if field in self.nolinkcheck:
-            return True
+            return link
```

Comment:
Can this be done somewhere/somehow else? This changes the return type from `bool` to `Union[bool, AnyStr, List[str], Dict[str, Any]]` which isn't adequately checked for elsewhere and is a bit messy :-)